gen/utils/data/class.go

```diff
@@ -0,0 +1,202 @@
+	if err != nil {
+		genLogger.Fatal(fmt.Sprintf("Error during reading of file: %s.", err.Error()))
+	}
+
+	genLogger.Trace(fmt.Sprintf("Parsing meta file for class '%s'.", c.ClassName))
+	// For now, the file content is unmarshalled into a map[string]interface{} and then set the class data.
+	// This is done because we add logic on top of the file content to set the class data.
+	// ENHANCEMENT: investigate if we can unmarshal the file content directly into a class struct specific for meta.
+	var metaFileContent map[string]interface{}
+	err = json.Unmarshal(fileContent, &metaFileContent)
+	if err != nil {
+		genLogger.Fatal(fmt.Sprintf("Error during Unmarshal(): %s.", err.Error()))
+	}
+
+	c.MetaFileContent = metaFileContent[fmt.Sprintf("%s:%s", c.ClassNamePackage, c.ClassNameShort)].(map[string]interface{})
+
+	genLogger.Debug(fmt.Sprintf("Succesfully loaded meta file for class '%s'.", c.ClassName))
+}
+
+func (c *Class) setClassData() {
+	genLogger.Debug(fmt.Sprintf("Setting class data for class '%s'.", c.ClassName))
+
+	c.setResourceName()
+
+	// TODO: add functions to set the other class data
+
+	if properties, ok := c.MetaFileContent["properties"]; ok {
+		c.setProperties(properties.(map[string]interface{}))
+	}
+
+	genLogger.Debug(fmt.Sprintf("Succesfully set class data for class '%s'.", c.ClassName))
+}
+
+func (c *Class) setResourceName() {
+	genLogger.Trace(fmt.Sprintf("Implement setting resourceName for class '%s'.", c.ClassName))
+}
+
+func (c *Class) setProperties(properties map[string]interface{}) {
+	genLogger.Debug(fmt.Sprintf("Setting properties for class '%s'.", c.ClassName))
+	for name, propertyDetails := range properties {
+		details := propertyDetails.(map[string]interface{})
+		// TODO: add logic to set the property data based on ignore/include/exclude overwrites (read-only) from definition files.
+		if details["isConfigurable"] == true {
+			c.Properties[name] = NewProperty(name, details)
+			c.PropertiesAll = append(c.PropertiesAll, name)
+			// TODO: add logic to set the required/optional/read-only list logic
+		}
+	}
+
+	genLogger.Debug(fmt.Sprintf("Succesfully set properties for class '%s'.", c.ClassName))
+	// TODO: add sorting logic for the properties
+}
+
+func splitClassNameToPackageNameAndShortName(className string) (string, string) {
+	// Splitting the class name into the package and short name.
+	// The package and short names are used for the meta file download, documentation links and lookup in the raw data.
+	var shortName, packageName string
+	genLogger.Trace(fmt.Sprintf("Splitting class name '%s' for name space separation.", className))
+	for index, character := range className {
+		if unicode.IsUpper(character) {
+			shortName = className[index:]
+			packageName = className[:index]
+			break
+		}
+	}
+
+	if packageName == "" || shortName == "" {
+		genLogger.Trace(fmt.Sprintf("Class name '%s' got split into package name '%s' and short name '%s'.", className, packageName, shortName))
```

Comment:
Isn't this supposed to be outside the if statement at the bottom?

Reply:
was intended to display only when it failed but then would suggest to move above the conditional to always print the line, since the fail case is fatal

Reply:
We should probably move it after the if condition and before the return statement?